### PR TITLE
feature/#157-market

### PR DESCRIPTION
**Pull request by Zeavee**
…s (not implented yet)

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/item/Healthpack.java [hunk 1]
```diff
@@ -16,6 +16,11 @@ public Healthpack(double healthPackAmount) {
         this.healthPackAmount = healthPackAmount;
     }
 
+    @Override
+    public Item createCopy() {
+        return new Healthpack(healthPackAmount);
```
Comment (varathar53) [inline app/src/main/java/ch/epfl/sdp/item/Healthpack.java:21]: add javadoc

File: app/src/main/java/ch/epfl/sdp/item/Market.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.Map;
+import java.util.Random;
+
+import ch.epfl.sdp.entity.EntityType;
+import ch.epfl.sdp.entity.Player;
+import ch.epfl.sdp.logic.RandomGenerator;
+import ch.epfl.sdp.map.Displayable;
+import ch.epfl.sdp.map.GeoPoint;
+import ch.epfl.sdp.map.MapsActivity;
+
```
Comment (varathar53) [inline app/src/main/java/ch/epfl/sdp/item/Market.java:16]: add javadoc for the classes and for the public methods

File: app/src/main/java/ch/epfl/sdp/item/Market.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    public GeoPoint getLocation() {
+        return super.getLocation();
+    }
+
+    @Override
+    public EntityType getEntityType() {
+        return super.getEntityType();
+    }
+
+    @Override
```
Comment (varathar53) [inline app/src/main/java/ch/epfl/sdp/item/Market.java:58]: once for what is it?

Comment (Zeavee) [reply app/src/main/java/ch/epfl/sdp/item/Market.java:58]: once is a boolean that permit to display an entity on the map only one time (if it does not move). This boolean comes from displayable, so the doc should be there

File: app/src/main/java/ch/epfl/sdp/item/Market.java (hunk 2 above)
Comment (Zeavee) [reply app/src/main/java/ch/epfl/sdp/item/Market.java:16]: I'll add Javadoc for the new public methods

File: app/src/main/java/ch/epfl/sdp/item/Healthpack.java [hunk 4]
```diff
@@ -16,6 +16,11 @@ public Healthpack(double healthPackAmount) {
         this.healthPackAmount = healthPackAmount;
     }
 
+    @Override
+    public Item createCopy() {
```
Comment (SkyRiS3s) [inline app/src/main/java/ch/epfl/sdp/item/Healthpack.java:20]: Why do you need to have a "createCopy" method  when you could have used obj.clone() ? I am not saying what you did is wrong, I am just curious why you chose this approach. The advantage I see with the Cloneable interface is that you might be able to make "Items.java" implement Cloneable and then override the clone method in each item class (that is the healthpack, the shrinker, the trap, ...).

Also, given that you return a healthpack, I would suggest renaming the method to "createHealthpackCopy()".

Comment (Zeavee) [reply app/src/main/java/ch/epfl/sdp/item/Healthpack.java:20]: Totally fine with this, I just forgot that clonable exists, I will quickly fix this

**Pull request closed without merging**